runtime/packages/core-modules/assembly/system.ts

```diff
@@ -58,5 +64,47 @@ export class System {
         const decodedHeader = Header.fromU8Array(result);
         return decodedHeader.result;
     }
+    /**
+     * Get the nonce value of the given AccountId
+     * The format of the key for storing nonce in the storage is:
+     * SCALE(AccountId) + SCALE("nonce")
+     * @param who nonce of this account
+     */
+    static accountNonce(who: AccountId): UInt64{
+        const nonceKey: u8[] = Utils.stringsToU8a([System.nonceKey]);
+        const value = Storage.get(who.getAddress().concat(nonceKey));
+        if(value.isSome()){
+            const decValue = Bytes.decodeCompactInt((<ByteArray>value.unwrap()).values);
+            return new UInt64(decValue.value);
+        }
+        else{
+            return new UInt64(0);
+        }
+    }
+
+    /**
+     * Increment nonce of this account
+     * @param who account
+     */
+    static incAccountNonce(who: AccountId): void{
+        const oldNonce = System.accountNonce(who);
+        const nonceKey: u8[] = Utils.stringsToU8a([System.nonceKey]);
+        const newNonce = new UInt64(oldNonce.value + 1);
+        Storage.set(who.getAddress().concat(nonceKey), newNonce.toU8a());
+    }
+
+    /**
+     * Verifies the message and signature of the extrinsic
+     * @param signature 
+     * @param msg message to be verified
+     * @param sender 
+     */
+    static verifySignature(signature: Signature, msg: u8[], sender: AccountId): bool{
```

Comment:
I think that we should introduce new `core-module` called `crypto` and put the function there. At some point we will have the other crypto related APIs, so a good thing imo is to add crypto module